nodejs/week2/app.js

```diff
@@ -0,0 +1,56 @@
+const { json } = require("express");
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+const documents = require("./documents.json");
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/search", (req, res) => {
+  console.log(JSON.stringify(req.query));
+  const query = req.query.q.toLowerCase();
+  if (!query) {
+    res.json(documents);
+  } else {
+    const filtereddocs = documents.filter((document) => {
+      const matchesValue =
+        document.value && document.value.toLowerCase().includes(query);
+      const matchesName =
+        document.name && document.name.toLowerCase().includes(query);
+      const matchDesc =
+        document.description &&
+        document.description.toLowerCase().includes(query);
+      return matchesName || matchDesc || matchesValue;
+    });
+    res.json(filtereddocs);
+  }
+});
+
```

Comment:
shouldn't this filter code filter specific data from the document generally, will it search and extract this data?
{ "id": 2, "type": "doc", "value": "hello world"} 
so instead of using the name, or description as a key,  using `Object.key ` might be useful.